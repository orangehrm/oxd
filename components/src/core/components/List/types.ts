interface Visible {
  visible?: boolean;
}

interface ListEmitEvent {
  type: string;
  identifier: string;
}

interface Tooltip {
  tooltip?: string;
  flow?: string;
}

interface Style {
  [key: string]: string | undefined;
}

interface Icon {
  iconName?: string;
  iconSize?: string;
  iconStyle?: string;
}

interface ButtonIcon extends Icon {
  iconName?: string;
}

interface ListButtonProps extends Tooltip, ButtonIcon {
  label?: string;
  displayType?: string;
  size?: string;
  disabled?: boolean;
}

interface ListButton {
  props: ListButtonProps;
  events?: ListEmitEvent[];
  class?: string;
  style?: Style;
}

interface ListAction extends ListButton, Visible {
  conditional?: boolean;
  type?: string;
}

interface ListQuickSearch extends ListButtonProps, Visible {
  placeholder?: string;
  clear?: boolean;
}

interface ListLeftPanelHeader extends Visible {
  button: ListButton;
}

interface SelectDropdownButton extends Icon {
  labelMini: string;
  doubleLineLabel: boolean;
  label: string;
  displayType: string;
  style: Style;
}

export interface ListLeftPanelBody extends Visible {
  button: SelectDropdownButton;
  dropdown: {
    style: Style;
  };
  buttonSecondary: {
    visible: boolean;
    icon: string;
    tooltip: string;
    style: string;
  };
}

interface CellProps {
  link?: string;
  target?: string;
  pillProperty?: string[];
  displayFormat?: string;
  locale?: string;
}

type CellRendererFunction = () => void;

interface ListLeftPanel {
  visible: boolean;
  header: ListLeftPanelHeader;
  body: ListLeftPanelBody;
  list: {
    visible: boolean;
    bubble: {
      visible: boolean;
    };
  };
}

interface ListTopBar extends Visible {
  listRecordCount: {
    singleTerm: string;
    multiTerm: string;
  };
  bulkActions: ListAction[];
  singleActions: ListAction[];
  quickSearch: ListQuickSearch;
}

interface ListTableHeaderCell extends Icon {
  name?: string;
  slot?: string;
  title?: string;
  style?: Style;
  cellType?: string;
  cellProps?: CellProps;
  cellRenderer?: string | CellRendererFunction;
  sortField?: string;
  initialSortOrder?: string;
}

export interface ListTable {
  leftPanel?: ListLeftPanel;
  topBar?: ListTopBar;
  headers?: ListTableHeaderCell[];
  selectable?: boolean,
  addable?: boolean,
  pagination?: {
    maxPageLimit: number;
  };
}
