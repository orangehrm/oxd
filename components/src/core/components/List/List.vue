<template>
  <div
    class="orangehrm-container recruitment-container"
    :class="{
      'table-sidebar-open':
        configurations.table.sidebar.visible && state.isSidePanelOpen,
    }"
  >
    <oxd-table-sidebar
      v-if="configurations.table.sidebar.visible"
      class="oxd-table-sidebar"
      :class="{'with-filters': configurations.table.topFilters.visible}"
      width="230px"
      :side-panel-list="sidePanelList"
      :header-visible="configurations.table.sidebar.header.visible"
      :body-visible="configurations.table.sidebar.body.visible"
      :list-visible="configurations.table.sidebar.list.visible"
      :bubble-visible="configurations.table.sidebar.list.bubble.visible"
      :button="configurations.table.sidebar.header.button"
      :selected-list-item-id="selectedListItem.id"
      @sidePanelList:onSelect="sidePanelListOnSelect"
      @side-panel:onToggle="toggleSidePanel"
      @sidePanelList:onHeaderBtnClick="sidePanelListOnHeaderBtnClick"
      :is-side-panel-open="state.isSidePanelOpen"
    >
      <template v-slot:sidePanelBody>
        <slot name="sidePanelBody"></slot>
      </template>
      <template v-slot:footer>
        <slot name="footer"></slot>
      </template>
    </oxd-table-sidebar>
    <div
      class="table-card-list-wrapper"
      :class="{'w-100': !state.isSidePanelOpen}"
    >
      <oxd-table-filter
        v-if="configurations.table.topFilters.visible"
        class="list-table-filter"
        :filter-title="filterTitle"
      >
        <template v-slot:actionOptions>
          <div
            v-for="(action, index) in configurations.table.topFilters.actions"
            :key="index"
          >
            <component
              :is="action.type"
              v-if="state.selectedItemIndexes.length > 0"
              v-bind="action.props"
              v-on="eventBinder(action.events)"
              :class="action.class"
              :style="action.style"
            />
          </div>
        </template>
        <template v-slot:toggleOptions>
          <oxd-quick-search
            :placeholder="'Search'"
            :clear="true"
            :createOptions="quickSearchOptions"
            :modelValue="state.selectedQuickSearch"
            @dropdown:modelValue="quickSearchSelect"
          >
            <template v-slot:iconSlot>
              <oxd-icon-button
                v-if="!state.selectedQuickSearch"
                name="oxd-search"
                display-type="label-info"
                class="quick-search-btn"
              ></oxd-icon-button>
            </template>
            <template v-slot:option="{data}">
              <oxd-profile-pic size="small" :imageSrc="data.imageSrc" />
              <span class="margin-left">{{ data.label }}</span>
            </template>
          </oxd-quick-search>
          <oxd-icon-button
            v-if="configurations.drawer.visible"
            name="funnel"
            display-type="label-info"
            class="btn-large"
            @click="showFilterDrawer"
          />
          <!-- <oxd-icon-button name="gear-fill" /> -->
          <oxd-button
            :size="'medium'"
            display-type="label-info"
            label="CSV"
            icon-name="file-earmark-spreadsheet"
            @click="exportBtn"
          />
        </template>
      </oxd-table-filter>

      <div class="oxd-card-table-wrapper">
        <oxd-card-table
          :selector="state.selector"
          :headers="configurations.table.headers"
          :items="listItems"
          :selectable="true"
          :clickable="false"
          :isDynamicCell="true"
          :class="oxdCardTableStyleClasses"
          v-model:selected="state.checkedItems"
          v-model:order="order"
          @update:order="tableSort"
          @update:selected="tableSelect"
          rowDecorator="oxd-table-decorator-card"
        />
        <oxd-pagination
          class="list-pagination"
          :length="paginationLength"
          v-model:current="state.currentPage"
          :max="maxPages"
          :pages-list="pagination.pages"
          :current="pagination.limit"
          @previous="previous"
          @next="next"
          @clickPage="clickPage"
          @onPerPageSelect="perPageSelect"
        />
      </div>
    </div>
    <!-- <oxd-drawer
      v-if="configurations.drawer.visible"
      :modal-state="state.modalState"
      :width="configurations.drawer.width"
      :height="configurations.drawer.height"
      :full-height="configurations.drawer.fullHeight"
      :sticky-footer="configurations.drawer.stickyFooter"
      :fixed="configurations.drawer.fixedPosition"
      :position="configurations.drawer.position"
      :ok-button="{
        label: configurations.drawer.footer.okButton.label,
        click: applySearch,
      }"
      :cancel-button="{
        click: closeDrawer,
      }"
      @drawer:click-outside="closeDrawer"
    >
      <template v-slot:header>
        <div v-if="configurations.drawer.header.visible" class="header">
          <slot name="drawerHeader">
            <div class="d-flex align-center justify-between">
              <h5>{{ configurations.drawer.header.title }}</h5>
              <oxd-icon-button
                v-if="configurations.drawer.header.charmButton.visible"
                :name="configurations.drawer.header.charmButton.icon"
                @click="resetSearch"
              />
            </div>
          </slot>
        </div>
      </template>
      <template v-slot:body>
        <slot name="drawerBody"></slot>
      </template>
    </oxd-drawer> -->
  </div>
</template>

<script lang="ts">
import CardTable from '@orangehrm/oxd/core/components/CardTable/CardTable.vue';
import TableFilter from '@orangehrm/oxd/core/components/TableFilter/TableFilter.vue';
import IconButton from '@orangehrm/oxd/core/components/Button/Icon.vue';
import Button from '@orangehrm/oxd/core/components/Button/Button.vue';
import QuickSearchInput from '@orangehrm/oxd/core/components/Input/Autocomplete/QuickSearchInput.vue';
import TableSidebar from '@orangehrm/oxd/core/components/TableSidebar/TableSidebar.vue';
import ProfilePic from '@orangehrm/oxd/core/components/ProfilePic/ProfilePic.vue';
// import Drawer from '@orangehrm/oxd/core/components/Drawer/Drawer.vue';
import Pagination from '@orangehrm/oxd/core/components/Pagination/Pagination.vue';
import images from './images';

import {defineComponent, reactive, computed} from 'vue';

export default defineComponent({
  components: {
    'oxd-card-table': CardTable,
    'oxd-table-sidebar': TableSidebar,
    'oxd-table-filter': TableFilter,
    'oxd-button': Button,
    'oxd-icon-button': IconButton,
    'oxd-quick-search': QuickSearchInput,
    'oxd-profile-pic': ProfilePic,
    // 'oxd-drawer': Drawer,
    'oxd-pagination': Pagination,
  },
  props: {
    configurations: {
      type: Object,
      required: true,
    },
    listItems: {
      type: Array,
    },
    filteredTotalRecordsCount: {
      type: Number,
      default: 0,
    },
    wholeRecordsCount: {
      type: Number,
      default: 0,
    },
    sidePanelList: {
      type: Array,
      default: () => [],
    },
    quickSearchOptions: {
      type: Function,
      required: true,
    },
    pagination: {
      type: Object,
      default: () => ({
        limit: 10 as number,
        pages: [10, 20, 50, 100] as number[],
      }),
    },
    selectedListItem: {
      type: Object,
      default: () => ({}),
    },
    maxPages: {
      type: Number,
      default: 5,
    },
  },
  setup(props, {emit}) {
    const sampleImages = images;

    const state = reactive({
      currentPage: 1,
      isSidePanelOpen: true,
      checkedItems: [],
      modalState: false as boolean,
      selectedQuickSearch: null,
      selectedItemIndexes: [],
    });

    const oxdCardTableStyleClasses = computed(() => {
      let styleClasses = 'oxd-classic-table ';
      styleClasses += props.configurations.table.topFilters.visible
        ? 'with-filters'
        : '';
      return styleClasses;
    });

    const order = computed(() => {
      const sortableFieldsObj = {};
      props.configurations.table.headers.forEach(header => {
        if (header.initialSortOrder) {
          sortableFieldsObj[header.sortField] = header.initialSortOrder;
        }
      });
      return sortableFieldsObj;
    });

    const isFloat = n => {
      return n === +n && n !== (n | 0);
    };

    const paginationLength = computed((): number => {
      const pagesLength: number =
        props.filteredTotalRecordsCount / props.pagination.limit;
      return isFloat(pagesLength) ? Math.floor(pagesLength) + 1 : pagesLength;
    });

    const filterTitle = computed((): string => {
      let title = '';
      if (state.selectedItemIndexes.length > 0) {
        if (state.selectedItemIndexes.length > 1) {
          title =
            props.configurations.table.topFilters.listRecordCount.multiTerm;
        } else {
          title =
            props.configurations.table.topFilters.listRecordCount.singleTerm;
        }
        title = `(${state.selectedItemIndexes.length}) ${title} selected`;
      } else {
        if (props.filteredTotalRecordsCount > 1) {
          title =
            props.configurations.table.topFilters.listRecordCount.multiTerm;
        } else {
          title =
            props.configurations.table.topFilters.listRecordCount.singleTerm;
        }
        title = `(${props.filteredTotalRecordsCount}) ${title} found`;
      }
      return title;
    });

    const sidePanelListOnHeaderBtnClick = () => {
      emit('sidePanelList:onHeaderBtnClick');
    };

    const sidePanelListOnSelect = item => {
      emit('sidePanelList:onSelect', item);
    };

    const quickSearchSelect = value => {
      if (value) {
        state.selectedQuickSearch = {
          label: value.candidateName,
        };
        emit('quick-search:onSelect', value);
      } else {
        state.selectedQuickSearch = null;
        emit('quick-search:onClear');
      }
    };

    const tableSort = value => {
      emit('update:order', value);
    };

    const tableSelect = items => {
      state.selectedItemIndexes = items;
      emit('update:selected', items);
    };

    const showFilterDrawer = (): void => {
      state.modalState = !state.modalState;
      emit('drawer:show', true);
    };

    const applySearch = (): void => {
      state.modalState = false;
      emit('drawer-search:apply');
    };

    const resetSearch = (): void => {
      state.modalState = false;
      emit('drawer-search:reset');
    };

    const closeDrawer = (): void => {
      state.modalState = false;
      emit('drawer-search:cancel');
    };

    const toggleSidePanel = (sidebarOpenState: boolean) => {
      state.isSidePanelOpen = sidebarOpenState;
      emit('table-sidebar:onToggle', sidebarOpenState);
    };

    const previous = (e: Event) => {
      emit('pagination:onPrevious', e);
    };

    const next = (e: Event) => {
      emit('pagination:onNext', e);
    };

    const clickPage = (page: number, e: Event) => {
      emit('pagination:onSelectPage', page, e);
    };

    const perPageSelect = (page: number) => {
      emit('pagination:onSelectPerPage', page);
    };

    const exportBtn = () => {
      emit('topfilters:onExportBtnClick')
    }

    const eventBinder = (events) => {
      let mappedEvents, mappedEventsObj
      if (events) {
        mappedEvents = events.map(event => {
          return {
            "click": (vals) => {
              debugger
              emit(event.identifier, vals)
            }
          }
        })
        mappedEventsObj = Object.assign({}, ...mappedEvents )
      }
      return mappedEventsObj;
    }

    return {
      state,
      filterTitle,
      sampleImages,
      oxdCardTableStyleClasses,
      order,
      sidePanelListOnHeaderBtnClick,
      sidePanelListOnSelect,
      quickSearchSelect,
      showFilterDrawer,
      applySearch,
      resetSearch,
      closeDrawer,
      toggleSidePanel,
      tableSort,
      tableSelect,
      previous,
      next,
      clickPage,
      perPageSelect,
      paginationLength,
      exportBtn,
      eventBinder,
    };
  },
});
</script>

<style lang="scss" scoped>
:deep(.header-image) {
  img {
    width: 2.5rem !important;
  }
}

.recruitment-container {
  margin-top: 2rem;
  background-color: #fff;
  border-radius: 1.2rem;
  display: flex;
  align-items: flex-start;
  .oxd-table-sidebar {
    &.with-filters {
      margin-top: 1.25rem;
    }
  }
  .table-card-list-wrapper {
    width: 100%;
    .list-table-filter {
      padding-bottom: 0;
      :deep(.oxd-divider) {
        margin-top: 0.25rem;
        margin-bottom: 0;
      }
      // :deep(.oxd-table-filter-header-title) {
      //   .oxd-table-filter-title {
      //     // Use Sub Title in OXD
      //     // font-weight: 500 !important;
      //   }
      // }
      :deep(.oxd-table-filter-header) {
        .oxd-table-filter-header-options {
          align-items: center;
          .oxd-icon-button {
            i {
              font-size: 14px;
            }
          }
          .oxd-button {
            display: flex;
            align-items: center;
            padding: 0.5rem 1rem !important;
            min-width: unset !important;
            font-size: 0.75rem;
            i {
              font-size: 1.0625rem;
              margin-right: 0.25rem;
            }
            &:hover,
            &:focus {
              color: #1e6ceb;
              background-color: rgba(30, 108, 235, 0.15);
            }
          }
        }
      }
      .oxd-autocomplete-search-wrapper {
        width: 300px;
      }
      :deep(.oxd-autocomplete-text-input) {
        background-color: #fff;
        padding-top: unset;
        padding-bottom: unset;
        height: 36px;
      }
      :deep(.quick-search-btn) {
        padding: 0;
        min-width: unset;
        min-height: unset;
        width: 28px;
        height: 28px;
        background-color: rgba(100, 114, 140, 0.1);
        i {
          width: unset;
          height: unset;
          svg {
            width: 12px;
          }
        }
      }
    }
  }
  &.table-sidebar-open {
    .table-card-list-wrapper {
      width: calc(100% - 230px);
    }
  }
  .oxd-classic-table {
    flex-wrap: nowrap;
    :deep(.oxd-icon-button) {
      font-size: 15px;
    }
    :deep(.oxd-table-th) {
      text-align: left;
      padding-top: 0.65rem;
      padding-bottom: 0.65rem;
    }
    :deep(.oxd-table-cell) {
      flex-wrap: nowrap;
    }
  }
}
.w-100 {
  width: 100% !important;
}
.margin-left {
  margin-left: 0.5rem;
}
.list-pagination {
  height: initial;
  line-height: initial;
  padding: 25px;
  display: flex;
  align-items: center;
  justify-content: flex-end;
  box-shadow: none;
  :deep(i) {
    font-size: initial;
    height: initial;
    line-height: initial;
  }
  :deep(.oxd-select-wrapper) {
    margin-right: 5px;
  }
}
.oxd-button {
  display: flex;
  align-items: center;
  :deep(.oxd-button-icon) {
    margin-right: 0.5rem;
  }
  &.btn-info {
    :deep(svg) {
      * {
        fill: #17a2b8 !important;
      }
    }
    color: #17a2b8;
    background-color: #327cf31a;
    &:hover {
      color: #17a2b8;
      background-color: rgba(30, 108, 235, 0.15);
      svg {
        * {
          fill: #fff !important;
        }
      }
    }
  }
  &.btn-danger {
    :deep(svg) {
      * {
        fill: #eb0910 !important;
      }
    }
    color: #eb0910;
    background-color: rgba(235, 9, 16, 0.1);
    &:hover {
      color: #eb0910;
      background-color: rgba(235, 9, 16, 0.15);
      svg {
        * {
          fill: #fff !important;
        }
      }
    }
  }
}
</style>
