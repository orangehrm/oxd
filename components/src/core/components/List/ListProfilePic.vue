<template>
  <oxd-profile-pix>
    <img :src="imageSrc" />
  </oxd-profile-pix>
</template>

<script>
import ProfilePic from '@orangehrm/oxd/core/components/ProfilePic/ProfilePic';
export default {
  props: {
    imageSrc: {
      type: String,
      default: '',
    },
  },
  components: {
    'oxd-profile-pix': ProfilePic,
  },
};
</script>
