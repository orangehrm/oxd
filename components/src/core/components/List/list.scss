@import '../../../styles';

.oxd-list-container {
  min-width: 65rem;
  margin-top: 1.5rem;
  background-color: $oxd-white-color;
  border-radius: 1.2rem;
  padding: 1rem;
  box-sizing: border-box;
  .oxd-table-left-panel {
    flex-shrink: 0;
    &.with-filters {
      margin-bottom: 0.25rem;
    }
    :deep(.oxd-table-left-panel--header) {
      display: flex;
      flex-direction: column;
      justify-content: space-between;
      .table-header-action-btn-container {
        margin-bottom: 0.5rem;
      }
      .table-header-action-btn {
        margin: 0 0 1rem 0;
      }
    }
    :deep(.oxd-table-left-panel--separator) {
      margin-top: 0;
      margin-bottom: 0;
    }
    &--toggle-btn {
      top: 42px;
    }
  }
  .table-card-list-wrapper {
    width: 100%;
    height: 100%;
    border-left: 1px solid $oxd-interface-gray-lighten-4-color;
    .list-table-filter {
      padding: 0 0 0 0;
      margin: 0 0 0 1rem;
      border-radius: 0;
      :deep(.oxd-table-filter-header) {
        min-height: 2.5rem;
        margin-bottom: 0.85rem;
        .oxd-table-filter-header-title {
          .oxd-table-filter-title {
            font-weight: 600;
            color: $oxd-interface-gray-color;
            font-size: $oxd-noemal-text-14;
          }
        }
      }
      &.items-selected {
        :deep(.oxd-table-filter-header) {
          .oxd-table-filter-header-title {
            .oxd-table-filter-title {
              font-weight: 700;
              color: $oxd-interface-gray-darken-1-color;
            }
          }
        }
      }
    }
  }
  &.table-left-panel-open {
    .table-card-list-wrapper {
      flex-grow: 1;
    }
  }
  .oxd-classic-table {
    padding-right: 0;
    padding-left: 0.625rem;
    padding-top: 0;
    flex-wrap: nowrap;
    :deep(.oxd-icon-button) {
      font-size: 15px;
    }
    :deep(.oxd-table-th) {
      &:first-child {
        padding-left: 0;
      }
      text-align: left;
      padding-top: 0.65rem;
      padding-bottom: 0.65rem;
      .oxd-table-header-sort {
        .oxd-svg-icon {
          &:hover {
            color: $oxd-interface-gray-darken-1-color;
          }
        }
      }
    }
    :deep(.oxd-table-cell) {
      &:first-child {
        padding-left: 0;
      }
      flex-wrap: nowrap;
      .oxd-select-wrapper {
        max-width: 195px;
        .oxd-select-text {
          max-width: 195px;
          width: -moz-fit-content;
          width: fit-content;
        }
      }
    }
    :deep(.oxd-checkbox-input) {
      @media (min-width: 0px) and (max-width: 1366px) {
        margin-right: 0;
      }
    }
  }
}
.margin-left {
  margin-left: 0.5rem;
}
.list-pagination {
  height: initial;
  line-height: initial;
  padding: 25px 0 25px 25px;
  box-shadow: none;
}
.oxd-button {
  display: flex;
  align-items: center;
  &:disabled {
    :deep(svg) {
      * {
        fill: $oxd-white-color !important;
      }
    }
  }
  &.oxd-btn-info-custom {
    :deep(svg) {
      * {
        fill: #17a2b8;
      }
    }
    color: #17a2b8;
    &:disabled {
      background-color: $oxd-interface-gray-lighten-1-color !important;
      color: $oxd-white-color !important;
    }
  }
  &.oxd-btn-danger-custom {
    :deep(svg) {
      * {
        fill: $oxd-feedback-danger-color;
      }
    }
    &:disabled {
      background-color: $oxd-interface-gray-lighten-1-color !important;
      color: $oxd-white-color !important;
    }
  }
}

.oxd-autocomplete-search-wrapper {
  :deep(.oxd-autocomplete-dropdown) {
    padding: 0.5rem 0.25rem 0.5rem 0.25rem;
  }
  :deep(.oxd-autocomplete-option) {
    padding: 0.25rem 0.75rem;
  }
}
