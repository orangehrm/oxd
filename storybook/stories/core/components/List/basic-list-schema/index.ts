import table from "./basic-table.list";

export default {
    table
};
