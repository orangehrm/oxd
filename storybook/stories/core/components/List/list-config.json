{
  "permission": {
    "create": true,
    "read": true,
    "update": true,
    "delete": true
  },
  "table": {
    "leftPanel": {
      "visible": true,
      "header": {
        "visible": true,
        "button": {
          "props": {
            "label": "Add Candidate",
            "displayType": "secondary",
            "size": "long",
            "display-type": "label-info",
            "icon-name": "plus",
            "iconSize": "small"
          }
        }
      },
      "body": {
        "visible": true,
        "button": {
          "label": "All Vacancy",
          "labelMini": "Vacancy",
          "iconName": "eye",
          "iconSize": "large",
          "iconStyle": "height: 20px",
          "displayType": "label",
          "doubleLineLabel": true
        },
        "dropdown": {
          "style": {
            "width": "220px"
          }
        }
      },
      "list": {
        "visible": true,
        "bubble": {
          "visible": true
        }
      }
    },
    "topBar": {
      "visible": true,
      "listRecordCount": {
        "singleTerm": "Candidate",
        "multiTerm": "Candidates"
      },
      "bulkActions": [
        {
          "conditional": true,
          "type": "oxd-button",
          "props": {
            "label": "Archive",
            "size": "medium",
            "display-type": "label-info",
            "icon-name": "oxd-archive"
          },
          "class": "oxd-btn-info-custom",
          "style": {
            "margin-left": "0.5rem"
          },
          "events": [
            {
              "type": "click",
              "identifier": "archive-candidates"
            }
          ]
        },
        {
          "conditional": true,
          "visible": false,
          "type": "oxd-button",
          "props": {
            "label": "Unarchive",
            "size": "medium",
            "display-type": "label-info",
            "icon-name": "oxd-archive"
          },
          "class": "oxd-btn-info-custom",
          "style": {
            "margin-left": "0.5rem"
          },
          "events": [
            {
              "type": "click",
              "identifier": "unarchive-candidates"
            }
          ]
        },
        {
          "type": "oxd-button",
          "props": {
            "label": "Delete Selected",
            "size": "medium",
            "display-type": "label-danger",
            "icon-name": "trash"
          },
          "class": "oxd-btn-danger-custom",
          "style": {
            "margin-left": "0.5rem"
          },
          "events": [
            {
              "type": "click",
              "identifier": "delete-candidates"
            }
          ]
        }
      ],
      "quickSearch": {
        "visible": true,
        "placeholder": "Search",
        "clear": true,
        "style": {
          "width": "261px"
        },
        "button": {
          "visible": true,
          "class": "quick-search-btn",
          "style": {
            "min-width": "unset",
            "min-height": "unset"
          },
          "props": {
            "name": "oxd-search",
            "displayType": "label-info",
            "size": "small"
          }
        }
      },
      "singleActions": [
        {
          "type": "oxd-icon-button",
          "props": {
            "size": "xx-small",
            "display-type": "label-info",
            "name": "oxd-filter",
            "tooltip": "Filter",
            "flow": "top"
          },
          "class": "btn-large",
          "style": {
            "margin-left": "font-size: 14px"
          },
          "events": [
            {
              "type": "click",
              "identifier": "drawer:show"
            }
          ]
        },
        {
          "type": "oxd-button",
          "props": {
            "label": "CSV",
            "size": "medium",
            "display-type": "label-info",
            "icon-name": "file-earmark-spreadsheet",
            "tooltip": "Export to CSV",
            "flow": "top"
          },
          "events": [
            {
              "type": "click",
              "identifier": "topfilters:onExportBtnClick"
            }
          ]
        }
      ]
    },
    "headers": [
      {
        "name": "profilePic",
        "slot": "footer",
        "title": "",
        "style": {
          "width": "50px",
          "justify-content": "center"
        },
        "cellType": "oxd-table-cell-profile-pic",
        "iconName": "oxd-profile-pic",
        "iconStyle": "color: #929baa;",
        "cellProps": {
          "link": "url2",
          "target": "_blank"
        }
      },
      {
        "name": "candidate",
        "sortField": "candidate",
        "initialSortOrder": "DEFAULT",
        "title": "Candidate",
        "style": {
          "flex": 1
        },
        "cellType": "oxd-table-cell-link-with-pill",
        "cellProps": {
          "link": "url1",
          "target": "_blank",
          "pillProperty": ["vacancy", "vacancyName"]
        }
      },
      {
        "name": "email",
        "sortField": "email",
        "initialSortOrder": "DEFAULT",
        "title": "Email",
        "style": {
          "flex": 1
        }
      },
      {
        "name": "contactNumber",
        "sortField": "contactNumber",
        "initialSortOrder": "DEFAULT",
        "title": "Contact Number",
        "style": {
          "flex": 1
        }
      },
      {
        "name": "dateOfApplication",
        "sortField": "dateOfApplication",
        "initialSortOrder": "DESC",
        "title": "Date Applied",
        "style": {
          "flex": 1
        },
        "cellType": "oxd-table-cell-date",
        "cellProps": {
          "displayFormat": "l, d-M-Y"
        }
      },
      {
        "name": "action",
        "slot": "footer",
        "title": "Stage",
        "style": {
          "flex": "1 1 200px"
        },
        "cellType": "oxd-table-cell-actions",
        "cellRenderer": "actionsRenderer"
      }
    ],
    "tdMapper": [
      {
        "name": "",
        "title": "",
        "style": "",
        "link": ""
      }
    ],
    "pagination": {
      "maxPageLimit": 5
    }
  },
  "drawer": {
    "visible": true,
    "width": "600px",
    "height": "100%",
    "fullHeight": false,
    "stickyFooter": true,
    "fixedPosition": true,
    "position": "right",
    "header": {
      "visible": true,
      "title": "Filter",
      "charmButton": {
        "visible": true,
        "icon": "oxd-refresh"
      }
    },
    "footer": {
      "okButton": {
        "label": "Search"
      }
    }
  }
}
