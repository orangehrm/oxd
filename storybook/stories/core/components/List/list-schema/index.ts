import table from "./table.list";
import drawer from "./drawer.table.list";
import modals from "./modals.table.list";

export default {
  table,
  drawer,
  modals,
};
