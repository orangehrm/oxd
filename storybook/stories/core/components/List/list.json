{
  "table": {
    "sidePanel": {
      "visible": true,
      "header": {
        "visible": true,
        "button": {
          "label": "Add Candidate"
        }
      },
      "body": {
        "visible": true
      },
      "list": {
        "visible": true,
        "bubble": {
          "visible": true
        }
      }
    },
    "topFilters": {
      "visible": true
    },
    "headers": [
      {
        "name": "profilepic",
        "slot": "footer",
        "title": "",
        "style": {
          "width": "44px"
        },
        "cellType": "oxd-table-cell-actions",
        "cellRenderer": "profilePicRenderer"
      },
      {
        "name": "candidate",
        "sortField": "candidate",
        "title": "Candidate",
        "style": {
          "flex": 1
        }
      },
      {
        "name": "email",
        "sortField": "email",
        "title": "Email",
        "style": {
          "flex": 1
        }
      },
      {
        "name": "contactNumber",
        "sortField": "contactNumber",
        "title": "Contact Number",
        "style": {
          "flex": 1
        }
      },
      {
        "name": "dateOfApplication",
        "sortField": "dateOfApplication",
        "title": "Date Applied",
        "style": {
          "flex": 1
        }
      },
      {
        "name": "action",
        "slot": "footer",
        "title": "Stage",
        "style": {
          "width": "300px"
        },
        "cellType": "oxd-table-cell-actions",
        "cellRenderer": "actionsRenderer"
      }
    ]
  }
}
