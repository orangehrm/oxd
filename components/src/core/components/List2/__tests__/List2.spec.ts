import {mount} from '@vue/test-utils';
import List from '@orangehrm/oxd/core/components/List2/List2.vue';
import IconButton from '@orangehrm/oxd/core/components/Button/Icon.vue';
import {h} from 'vue';

describe('List > List2.vue', () => {
  it('renders List', () => {
    const wrapper = mount(List);
    expect(wrapper.html()).toMatchSnapshot();
  });
  it('renders List with flat top', () => {
    const wrapper = mount(List, {
      props: {
        flatTop: true,
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
  it('collapse sidepanel on click', async () => {
    const wrapper = mount(List, {
      props: {
        showCollapse: true,
      },
      slots: {
        sidepanel: h('div', {}, 'Sidepanel'),
      },
    });
    expect(wrapper.vm.isCollapsed).toBeFalsy();
    wrapper.findComponent(IconButton).trigger('click');
    await wrapper.vm.$nextTick();
    expect(wrapper.vm.isCollapsed).toBeTruthy();
  });
});
