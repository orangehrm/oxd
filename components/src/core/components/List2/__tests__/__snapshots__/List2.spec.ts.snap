// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List > List2.vue renders List 1`] = `
<div class="oxd-sheet oxd-sheet--rounded oxd-sheet--white oxd-list-card">
  <!--v-if-->
  <!--v-if-->
  <div class="oxd-list-content">
    <div class="oxd-list-content-header">
      <div class="oxd-list-content-header --left"></div>
      <div class="oxd-list-content-header --right"></div>
    </div>
    <div>
      <hr class="oxd-divider-horizontal" role="separator" aria-orientation="horizontal">
      <!--v-if-->
    </div>
    <div class="oxd-list-content-body"></div>
    <!--v-if-->
    <div class="oxd-list-content-footer"></div>
  </div>
</div>
`;

exports[`List > List2.vue renders List with flat top 1`] = `
<div class="oxd-sheet oxd-sheet--rounded oxd-sheet--white oxd-list-card --flat-top">
  <!--v-if-->
  <!--v-if-->
  <div class="oxd-list-content">
    <div class="oxd-list-content-header">
      <div class="oxd-list-content-header --left"></div>
      <div class="oxd-list-content-header --right"></div>
    </div>
    <div>
      <hr class="oxd-divider-horizontal" role="separator" aria-orientation="horizontal">
      <!--v-if-->
    </div>
    <div class="oxd-list-content-body"></div>
    <!--v-if-->
    <div class="oxd-list-content-footer"></div>
  </div>
</div>
`;
